docs/containers/debug-node.md

```diff
@@ -124,10 +115,107 @@ Also note that the `debug` logger will not write logs unless enabled via the `DE
+
+The corresponding `uriFormat` in the debug launch configuration (in `launch.json`) to open the `about.html` page instead of the main page would be:
+
+```json
+{
+    "configurations": [
+        {
+            "name": "Docker Node.js Launch and Attach",
+            "type": "docker",
+            "request": "launch",
+            "preLaunchTask": "docker-run: debug",
+            "platform": "node",
+            "dockerServerReadyAction": {
+                "uriFormat": "http://localhost:%s/about.html"
+            }
+        }
+    ]
+}
+```
+
+## Mapping Docker container source files to the local workspace
+
+By default, the Docker extension assumes the application source files in the running Docker container are located in an `/usr/src/app` folder, and the debugger then maps those files back to the root of the opened workspace, in order to translate breakpoints from the container back to Visual Studio Code.
+
+If the application source files are in a different location (e.g. different Node.js frameworks have different conventions), either within the Docker container or within the opened workspace, then one or both of the `localRoot` and `remoteRoot` properties of the [`node`](debug-node.md#node-object-properties) object of the debug launch configuration should be set the root source locations within the workspace and the Docker container, respectively.
+
+For example, if the application instead resides in `/usr/my-custom-location`, the corresponding `remoteRoot` property would be:
+
+```json
+{
+    "configurations": [
+        {
+            "name": "Docker Node.js Launch and Attach",
+            "type": "docker",
+            "request": "launch",
+            "preLaunchTask": "docker-run: debug",
+            "platform": "node",
+            "node": {
+                "remoteRoot": "/usr/my-custom-location
+            }
+        }
+    ]
+}
+```
+
+# Troubleshooting
+
+## Problem: Docker image fails to build or start due to missing `node_modules` folder
 
-### Mapping Docker container source files to the local workspace
+Dockerfiles are often arranged in such a way as to optimize either image build time, image size, or both.  However, not every Node.js application frameworks supports all of the typical Node.js Dockerfile optimizations.  In particular, some frameworks the `node_modules` folder be an immediate subfolder of the application root folder, whereas, the Docker extension scaffolds a Dockerfile where the `node_modules` folder exists at a parent or ancestor level (which is generally allowed by Node.js).
```

Comment:
not every Node.js application framework (singular)

Reply:
"in some frameworks, the `node_modules` folder is ...

Reply:
Updated.